File: 2683-neighboring-bitwise-xor/2683-neighboring-bitwise-xor.py
class Solution:
    def doesValidArrayExist(self, derived: List[int]) -> bool:
        # Create an original array initialized with 0.
        original, n = [0], len(derived)
        for i in range(n):
            original.append(derived[i] ^ original[i])

        # Store the validation results in check_for_zero and check_for_one respectively
        check_for_zero = original[0] == original[-1]
        original = [1]
        for i in range(n):
            original.append(derived[i] ^ original[i])
        check_for_one = original[0] == original[-1]

        return check_for_zero or check_for_one

File: 2683-neighboring-bitwise-xor/2683-neighboring-bitwise-xor.cpp
class Solution {
public:
    bool doesValidArrayExist(vector<int>& derived) {
        // Create an original array initialized with 0.
        int n = derived.size();
        vector<int> original{0};
        for (int i = 0; i < n; ++i)
            original.push_back(original[i] ^ derived[i]);

        // Store the validation results in checkForZero and checkForOne respectively.
        bool checkForZero = (original[0] == original.back());
        original = {1};
        for (int i = 0; i < n; ++i)
            original.push_back(original[i] ^ derived[i]);

        bool checkForOne = (original[0] == original.back());

        return checkForZero | checkForOne;
    }
};